packages/design-system/src/components/combobox.tsx

```diff
@@ -117,11 +113,12 @@ export const ComboboxPopperContent = PopperContent;
 type ComboboxProps<Item> = {
   name: string;
   label?: string;
-  items: Array<Item>;
+  placeholder?: string;
+  items: ReadonlyArray<Item>;
   value?: Item;
-  onItemSelect?: (value: Item) => void;
+  onItemSelect?: (value?: Item) => void;
   onItemHighlight?: (value?: Item) => void;
-  itemToString?: (item: Item | null) => string;
+  itemToString?: (item?: Item | null) => string;
```

Comment:
do you really want to support both: null and undefined? Usually I am trying to support undefined only, because it works as an optional value, supporting both is usually not great from logic perspective

Reply:
This is mostly done to align with downshift types. I'm not a fan of both either and tried to make it simpler but couldn't succeed. Give it a try if you are fancy.